src/PrimitiveObsession/EngineBuilder.cs

```diff
@@ -1,30 +1,16 @@
-﻿using System.Text;
-using PrimitiveObsession.Config;
+﻿using PrimitiveObsession.Config;
 
 namespace PrimitiveObsession
 {
     public class EngineBuilder
     {
-        private readonly TireBuilder _tireBuilder;
-        private readonly PistonCount _pistonCount;
+        private readonly TireBuilder _tirebuilder;
+        public PistonCount PistonCount { get; }
 
-        public EngineBuilder(TireBuilder tireBuilder, PistonCount pistonCount)
+        public EngineBuilder(TireBuilder tirebuilder, PistonCount pistonPistonCount)
         {
-            _tireBuilder = tireBuilder;
-            _pistonCount = pistonCount;
-        }
-
-        public string Build()
-        {
-            var sb = new StringBuilder();
-
-            sb.Append("Pistons: ");
-            for (var i = 0; i < _pistonCount; i++) sb.Append("||");
-
-            sb.AppendLine();
-            sb.Append(_tireBuilder.Build());
-
-            return sb.ToString();
+            _tirebuilder = tirebuilder;
+            PistonCount = pistonPistonCount;
```

Comment:
Secondo me non ci serve che le classi facciano qualcosa. Al lettore non interessa. Al lettore interessa come facciamo ad iniettare le dipendenze primitive, non che calcoli ci facciamo dopo.

E qui in effetti c'è il rischio che il lettore cerchi di capire cosa vuoi tirare fuori con quelle concatenazioni di stringhe. Che poi sono un po' qui in questa classe e un po' in `TireBuilder`.
Se il lettore si mette a interpretare quel codice, toglie il focus da quello che vogliamo dire.
Se ignora quel codice, beh, meglio, ma allora perché lo abbiamo messo?

Io propongo di rimuoverlo. Avremo classi che non fanno nulla. Rimuoverei anche il codice dal costruttore: una volta che la classe ha ricevuto le sue dipenenze, a posto, non gli farei fare nient'altro.


Dal momento che non c'è più business logic, che ci sia 1 parametro iniettato, 2 o 10, cambia poco, no?
Infatti, se guardi il codice dei due parametri, `TireBuilder` e `PistonCount`, sono identici.
Io ne terrei solo uno. A che pro mostrare due volte la stessa cosa?